envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/GroupChangeWatcher.kt

```diff
@@ -43,10 +43,11 @@ internal class GroupChangeWatcher(
         ads: Boolean,
         request: DiscoveryRequest,
         knownResourceNames: MutableSet<String>,
-        responseConsumer: Consumer<Response>
-    ): Watch {
+        responseConsumer: Consumer<Response>,
+        hasClusterChanged: Boolean
+    ): Watch? {
```

Comment:
Why did return type changed? I guess because that's Java interface and we don't know if it's nullable?

Reply:
My bad. I think Intellij generated this and I didn't notice. I'll fix this.